lib/vector-core/buffers/src/internal_events.rs

```diff
@@ -0,0 +1,42 @@
+use core_common::internal_event::InternalEvent;
+use metrics::counter;
+
+pub struct BufferEventsReceived {
+    pub count: usize,
+    pub byte_size: usize,
+}
+
+impl InternalEvent for BufferEventsReceived {
+    fn emit_logs(&self) {}
```

Comment:
I think you can omit these. The `InternalEvent` trait implements stubs for them.